### ISSUE TITLE
base and expX

### ISSUE THREAD

**Issue by zhuang5252** [andrewpatrickdu/adversarial-yolov3-cowc#2]
What are the functions of base and expX?
`patch_configs = {
    "base": BaseConfig,
    "expX": ExperimentX,
    "sidestreet": SideStreet,
    "carpark": CarPark
}`

**Comment by andrewpatrickdu** [comment]
"base" is used to select config and weight files or YOLOv3, and to select the learning rate and schedule for training a patch.

 "expX" is used to run your own experiments (or train your own patch) by selecting your own parameters.

**Comment by zhuang5252** [comment]
Thank you. How is this document trained? And which script was used for training?
`yolov3-cowc_best_256.weights`

**Comment by andrewpatrickdu** [comment]
To train a patch, run the command:

python train_patch.py {experiment} {folder_name} 

e.g., 
python train_patch.py sidestreet experiment01

**Comment by zhuang5252** [comment]
I see what you mean, but how to train a better yolov3 weights ?
e.g 
Use `yolov3.weights` to detect the clean data(carpack all cars), i want to train the weight.

**Comment by andrewpatrickdu** [comment]
To obtain your own trained weights, use this github repo: https://github.com/AlexeyAB/darknet

**Comment by zhuang5252** [comment]
`yolov3-cowc_best_256.weights` is used cowc data and  this github repo: https://github.com/AlexeyAB/darknet to train. Is my understanding correct ? Can you provide your own training code ?

**Comment by andrewpatrickdu** [comment]
That's correct! I followed the instructions of that github repo to trained YOLOv3 on COWC-M. It's well written, so you shouldn't  have any problems.

**Comment by zhuang5252** [comment]
Could you please provide these two documents?
train  = /media/andrew/OS/COWC/cowc-m/datasets/DetectionPatches_256x256/train.txt
valid  = /media/andrew/OS/COWC/cowc-m/datasets/DetectionPatches_256x256/test.txt

**Comment by andrewpatrickdu** [comment]
[test.txt](https://github.com/andrewpatrickdu/adversarial-yolov3-cowc/files/9488066/test.txt)
[train.txt](https://github.com/andrewpatrickdu/adversarial-yolov3-cowc/files/9488071/train.txt)

**Comment by zhuang5252** [comment]
Is the dataset in the two files non-existent in the other file (wacv.tar.gz) ?

**Comment by andrewpatrickdu** [comment]
No, but you can download the COWC-M dataset from https://gdo152.llnl.gov/cowc/

**Comment by isher1895** [comment]
Excuse me, which file did you train YOLOv3 on COWC-M？

**Comment by isher1895** [comment]
Excuse me, in which file did you use COWC-M to train YOLOv3？
